src/Sparrow/Platform/ThreadMethods.cs

```diff
@@ -0,0 +1,155 @@
+﻿using System;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Threading;
+using NLog;
+
+namespace Sparrow
+{
+    public static class ThreadMethods
+    {
+        private static readonly Logger _log = LogManager.GetLogger(nameof(ThreadMethods));
+        private const int SCHED_OTHER = 0;
+        private const int SCHED_RR = 2;
+
+        public unsafe static void SetThreadPriority(ThreadPriority priority)
+        {
+            if (Platform.RunningOnPosix)
+            {
+                ulong threadId = PosixThreadsMethods.pthread_self();
+		        sched_param param = new sched_param
+		        {
+		            sched_priority = (int)priority
+		        };
+
+                int policy = SCHED_OTHER;
+                if (priority != ThreadPriority.Normal)
+                {
+                    policy = SCHED_RR;
+                    param.sched_priority = FixPosixPriority(priority);
+                }
+
+                var success = PosixThreadsMethods.pthread_setschedparam(threadId, policy, &param);
+                if (success != 0)
+                {
+                    int lastWin32ErrorCode = Marshal.GetLastWin32Error();
+                    throw new Win32Exception("Failed to set priority to thread " + threadId,
+                        new Win32Exception(lastWin32ErrorCode));
+                }
+            }
+            else
+            {
+                uint threadId = Win32ThreadsMethods.GetCurrentThreadId();
+                IntPtr handle = Win32ThreadsMethods.OpenThread(ThreadAccess.SetInformation, false, threadId);
```

Comment:
The pthread_setschedparam always return zero or none zero in the case the method fail